apps/server/src/middlewares/validateRequestBody.ts

```diff
@@ -0,0 +1,16 @@
+import { z } from 'zod';
+import { Request, Response, NextFunction } from 'express';
+
+export const validateRequestBody = (requestBody: any) => {
+    return (req: Request, res: Response, next: NextFunction) => {
+        try {
+            const schema = z.object(requestBody);
+            schema.parse(req.body);
+            return next();
+        } catch (err: any) {
+            return res
```

Comment:
Please first check if the error is really an instance of z.ZodError and then send the 400. Else you should send a 500. Plus have a look at how you can safely parse zod error messages as a simple err.message wont work all the time.